app/helpers/application_helper.rb

```diff
@@ -765,6 +765,26 @@ def _toolbar_chooser
     )
   end
 
+  # Calculate hash of toolbars to render
+  #
+  # keys are toolbar <div> names and values are toobar identifiers (now YAML files)
+  #
+  def calculate_toolbars
+    toolbars = {}
+    if inner_layout_present? # x_taskbar branch
+      toolbars['history_tb'] = history_toolbar_filename
+    elsif display_back_button? # taskbar branch
+      toolbars['summary_center_tb'] = controller.send(:restful?) ? "summary_center_restful_tb" : "summary_center_tb"
```

Comment:
`controller.restful?` should work just as well now, both definitions are public.